File: Concesionario/src/gestorAplicacion/activos/TransaccionModificacion.java
package gestorAplicacion.activos;

import java.io.Serializable;
import java.util.ArrayList;

import gestorAplicacion.personal.*;

public class TransaccionModificacion extends Transaccion implements Serializable{
	private static final long serialVersionUID = 1L;
	private Auto auto;
	private Mecanico mecanico;
	private Vendedor vendedor;
	static ArrayList<TransaccionModificacion> transaccionesmod = new ArrayList<TransaccionModificacion>();
	
	public TransaccionModificacion(String tipo, long ingreso, Cliente cliente, Auto auto, Mecanico mecanico, Articulo articulo, int transfer) {
		super(tipo, ingreso, cliente, transfer);
		this.auto=auto;
		this.mecanico=mecanico;
		this.articulo=articulo;
		TransaccionModificacion.transaccionesmod.add(this);
		
	}
	public TransaccionModificacion(String tipo, long ingreso, Cliente cliente, Auto auto, Vendedor vendedor, Articulo articulo, int transfer) {
		super(tipo, ingreso, cliente, transfer);
		this.auto=auto;
		this.setVendedor(vendedor);
		this.articulo=articulo;
		TransaccionModificacion.transaccionesmod.add(this);
		
	}
	

	
	@Override
	public String info() {
		 String txt = String.format("Transacción #%08d: venta realizada por %s para el cliente %s que adquirio el producto: %s de la marca: %s , con precio total $%d",
		            transfer, vendedor.getNombre(), cliente.getNombre(), articulo.getTipoArticulo(), articulo.getMarca(),articulo.getPrecio());
		return txt;
	}
	public String info2() {
		 String txt = String.format("Transacción #%08d: taller realizado por %s para el cliente %s que adquirio el producto: %s de la marca: %s , con precio total $%d",
		            transfer, mecanico.getNombre(), cliente.getNombre(), articulo.getTipoArticulo(), articulo.getMarca(),articulo.getPrecio());
		return txt;
	}
	
		
	public static Cliente getClientePorCedula(long cedula){
		Transaccion finder = null;
		Cliente client=null;
		
		for(Transaccion trans: transaccionesmod) {
			if(trans.getClienteCed()== cedula) {
				finder = trans;
				client=finder.getCliente();
				break;
			}
		}
		return client;
	}
	
	public static Auto getTransaccionporCedula(long cedula) {
		Auto finder = null;
		for (TransaccionModificacion trans: transaccionesmod) {
			if(trans.getClienteCed()== cedula) {
				finder = trans.getAuto();
				break;
			}
		}
		return finder;
	}
	
	
	public void setAuto(Auto auto) {
	    this.auto = auto;
	}

	public Auto getAuto() {
	    return auto;
	}
	public void setMecanico(Mecanico mecanico) {
	    this.mecanico = mecanico;
	}

	public Mecanico getMecanico() {
	    return mecanico;
	}
	public void setArticulo(Articulo articulo) {
	    this.articulo = articulo;
	}

	public Articulo getArticulo() {
	    return articulo;
	}
	public static void agregarTransaccion(TransaccionModificacion transaccion) {
	    transaccionesmod.add(transaccion);
	}

	public static void eliminarTransaccion(TransaccionModificacion transaccion) {
	    transaccionesmod.remove(transaccion);
	}

	public static ArrayList<TransaccionModificacion> getTransaccionesmod() {
	    return transaccionesmod;
	}
	public Vendedor getVendedor() {
		return vendedor;
	}
	public void setVendedor(Vendedor vendedor) {
		this.vendedor = vendedor;
	}

	
	
}

File: PYTHON/Concesionario/gestorAplicacion/Activos/TransaccionModificacion.py
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)
from transaccion import Transaccion

class TransaccionModificacion(Transaccion):
    transaccionesmod = []

    def __init__(self, tipo, ingreso, cliente, auto, articulo, transfer, mecanico=None,vendedor=None):
        super().__init__(tipo, ingreso, cliente, transfer)
        self.auto = auto
        self.mecanico = mecanico
        self.vendedor = vendedor
        self.articulo = articulo
        TransaccionModificacion.transaccionesmod.append(self)


    def info(self):
        txt = f"Transacción #{self.transfer:08d}: venta realizada por {self.vendedor.nombre} para el cliente {self.cliente.nombre} que adquirió el producto: {self.articulo.tipoArticulo} de la marca: {self.articulo.marca}, con precio total ${self.articulo.precio}"
        return txt

    def info2(self):
        txt = f"Transacción #{self.transfer:08d}: taller realizado por {self.mecanico.nombre} para el cliente {self.cliente.nombre} que adquirió el producto: {self.articulo.tipoArticulo} de la marca: {self.articulo.marca}, con precio total ${self.articulo.precio}"
        return txt

    @staticmethod
    def get_clientePorCedula(cedula):
        finder = None
        client = None

        for trans in TransaccionModificacion.transaccionesmod:
            if trans.get_clienteCed() == cedula:
                finder = trans
                client = finder.get_cliente()
                break

        return client

    @staticmethod
    def get_transaccionporCedula(cedula):
        finder = None

        for trans in TransaccionModificacion.transaccionesmod:
            if trans.get_clienteCed() == cedula:
                finder = trans.get_auto()
                break

        return finder

    def set_auto(self, auto):
        self.auto = auto

    def get_auto(self):
        return self.auto

    def set_mecanico(self, mecanico):
        self.mecanico = mecanico

    def get_mecanico(self):
        return self.mecanico

    def set_articulo(self, articulo):
        self.articulo = articulo

    def get_articulo(self):
        return self.articulo

    @staticmethod
    def agregar_transaccion(transaccion):
        TransaccionModificacion.transaccionesmod.append(transaccion)

    @staticmethod
    def eliminar_transaccion(transaccion):
        TransaccionModificacion.transaccionesmod.remove(transaccion)

    @staticmethod
    def get_transaccionesmod():
        return TransaccionModificacion.transaccionesmod
    @staticmethod
    def set_transaccionesmod(qq):
        TransaccionModificacion.transaccionesmod=qq

    def get_vendedor(self):
        return self.vendedor

    def set_vendedor(self, vendedor):
        self.vendedor = vendedor